examples/baseball.py

```diff
@@ -147,29 +143,47 @@ def get_site_stats(array, player_names):
     Return the summarized statistics for a given array corresponding
     to the values sampled for a latent or response site.
     """
+    # TODO: only use pandas (or any lightweight tabular package) to display final result
     if len(array.shape) == 1:
         df = pd.DataFrame(array).transpose()
     else:
         df = pd.DataFrame(array, columns=player_names).transpose()
     return df.apply(pd.Series.describe, axis=1)[["mean", "std", "25%", "50%", "75%"]]
 
 
-def summary(traces, sites, player_names, transforms={}):
+def summary(trace_posterior, sites, player_names, transforms={}, diagnostics=True):
     """
     Return summarized statistics for each of the ``sites`` in the
     traces corresponding to the approximate posterior.
     """
-    marginal = EmpiricalMarginal(traces, sites).get_samples_and_weights()[0].numpy()
+    marginal = EmpiricalMarginal(trace_posterior, sites).get_samples_and_weights()[0]
+
+    if diagnostics and trace_posterior.sampler.num_chains > 1:
+        chain_indices = trace_posterior.sampler.chain_indices
+        n_eff, r_hat = chain_diagnostics(marginal, chain_indices)
+
     site_stats = {}
     for i in range(marginal.shape[1]):
         site_name = sites[i]
         marginal_site = marginal[:, i]
         if site_name in transforms:
             marginal_site = transforms[site_name](marginal_site)
-        site_stats[site_name] = get_site_stats(marginal_site, player_names)
+        site_stats[site_name] = get_site_stats(marginal_site.numpy(), player_names)
+        if diagnostics and trace_posterior.sampler.num_chains > 1:
+            site_stats[site_name] = site_stats[site_name].assign(n_eff=n_eff[i].numpy(),
+                                                                 r_hat=r_hat[i].numpy())
     return site_stats
 
 
+def chain_diagnostics(samples, chain_indices):
+    assert samples.size(0) == chain_indices.numel()
+    chain_samples = samples[chain_indices.reshape(-1)].reshape(
+        chain_indices.shape + samples.shape[1:])
+    n_eff = effective_sample_size(chain_samples, sample_dim=1, chain_dim=0)
+    r_hat = split_gelman_rubin(chain_samples, sample_dim=1, chain_dim=0)
```

Comment:
@neerajprad to avoid confusion, should we make chain_dim=0, sample_dim=1 by default in `stats`?

Reply:
I think so too - I think that would be a better convention to choose.

Reply:
I'll do it in #1525 if this pull request is fine in its own. Otherwise, I will make the change here too. :)

Reply:
Sure, let us address it in #1525.